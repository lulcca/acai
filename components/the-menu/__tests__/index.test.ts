import { beforeEach, describe, expect, it } from 'vitest';
import TheMenu from '@/components/the-menu/index.vue';
import { mountSuspended } from '@nuxt/test-utils/runtime';

describe('Components > TheMenu', () => {
  let wrapper: ReturnType<typeof mountSuspended>;

  beforeEach(async () => wrapper = await mountSuspended(TheMenu));

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render all buttons correctly', () => {
    const anchors = wrapper.findAll('a');
    expect(anchors.length).toBe(3);
    expect(anchors[0].text()).toMatch(/roleta/i);
    expect(anchors[1].text()).toMatch(/regras/i);
  });

  it('should contain the correct repository link', () => {
    const link = wrapper.find('[aria-label="repositorio"]');
    expect(link.exists()).toBe(true);
    expect(link.attributes('href')).toBe('https://github.com/lulcca/acai');
    expect(link.attributes('target')).toBe('_blank');
  });
});
