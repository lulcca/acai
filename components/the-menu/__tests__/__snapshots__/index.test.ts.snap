// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > TheMenu > should match snapshot 1`] = `
"<nav class="gap-4 grid h-full place-items-center w-full [grid-template-areas:'A_B_._C_D'] [grid-template-columns:auto_auto_1fr_auto_auto]"><a class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 px-3.5 py-2.5 ring-1 ring-inset ring-current disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 text-fuchsia-100 dark:text-fuchsia-900 hover:bg-primary dark:hover:bg-primary inline-flex items-center font-major uppercase [grid-area:A]" aria-label="roleta" href="#roleta">
    <!--v-if--><span class="">roleta</span>
    <!--v-if-->
  </a><a class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 px-3.5 py-2.5 ring-1 ring-inset ring-current disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 text-fuchsia-100 dark:text-fuchsia-900 hover:bg-primary dark:hover:bg-primary inline-flex items-center font-major uppercase [grid-area:B]" aria-label="regras" href="#regras">
    <!--v-if--><span class="">regras</span>
    <!--v-if-->
  </a><a class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 p-2.5 ring-1 ring-inset ring-current disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 text-fuchsia-100 dark:text-fuchsia-900 hover:bg-primary dark:hover:bg-primary inline-flex items-center [grid-area:C]" aria-label="repositorio" href="https://github.com/lulcca/acai" rel="noopener noreferrer" target="_blank"><span class="iconify i-mdi:github flex-shrink-0 h-6 w-6" aria-hidden="true"></span>
    <!--v-if-->
    <!--v-if-->
  </a><button type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 p-2.5 ring-1 ring-inset ring-current disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 text-fuchsia-100 dark:text-fuchsia-900 hover:bg-primary dark:hover:bg-primary inline-flex items-center [grid-area:D]"><span class="iconify i-heroicons:arrow-path-20-solid flex-shrink-0 h-6 w-6 animate-spin" aria-hidden="true"></span>
    <!--v-if-->
    <!--v-if-->
  </button></nav>"
`;
